data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/QueryParameterConfiguration.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.opensearch.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+
+public class QueryParameterConfiguration {
+
```

Comment:
Please add the `query_string` parameter here

Reply:
@ShalakaJain , OpenSearch has a query language which we should expect users to be familiar with. It seems to me that this configuration is attempting to mimic the OpenSearch query, but in YAML. Am I misunderstanding your intent here?

If we do try to mimic the OpenSearch query language we may hit some issues:

* We will have to keep our query up-to-date with changes in OpenSearch.
* Users are going to have to learn the Data Prepper form rather than use something they already know.